### PR TITLE
change failure logic to match new rejection feature

### PR DESCRIPTION
**Pull request by shilendras**

### CODE REVIEW

File: submission.json [hunk 1]
```diff
@@ -21,10 +21,8 @@
         "id": 3361,
         "name": "Accepted",
         "max_grade": 2,
-        "pass_grade": 2,
         "grade_labels": [
-          { "grade": 1, "label": "No" },
-          { "grade": 2, "label": "Yes" }
+          { "grade": 1, "label": "Yes" }
```
Comment (bodhish) [inline submission.json:25]: The hash is missing a grade label for `grade_2`. As we have given `2` as the `max_grade` I think its good we add grade label for `2`.

File: app/pupilfirst_api.rb [hunk 2]
```diff
@@ -56,18 +56,18 @@ def valid_status?(status)
...
-        }
+      if status == 'passed'
+        return @submission.evaluation_criteria.map do |criteria|
+          {
+            evaluationCriterionId: criteria['id'],
+            grade: criteria['max_grade']
+          }
+        end
+      else
+        return []
```
Comment (bodhish) [inline app/pupilfirst_api.rb:73]: I am not sure if we should send in an empty array when we could possibly ignore sending the filed altogether for failed submissions.

**Review by bodhish — CHANGES_REQUESTED** [review]

File: submission.json (hunk 1 above)
Comment (shilendras) [reply submission.json:25]: Missed that. Changed `max_grade` to 1. As that would be the default once the failure grades get removed.

File: app/pupilfirst_api.rb (hunk 2 above)
Comment (shilendras) [reply app/pupilfirst_api.rb:73]: I was sticking to the way it was written in lms. But I agree. Made the change in lms and grade-action as well.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -10,6 +10,7 @@ env:
   OPENAI_ORGANIZATION_ID: ${{ secrets.OPENAI_ORGANIZATION_ID }}
   REVIEW_BOT_USER_TOKEN: ${{ secrets.REVIEW_BOT_USER_TOKEN }}
   REVIEW_END_POINT: ${{ secrets.REVIEW_END_POINT }}
+  TEST_MODE: true
```
Comment (shilendras) [inline .github/workflows/test.yml:13]: @bodhish Shouldn't we put test_mode to true here? It is on grade-action.

**Comment by harigopal** [comment]
Before merging this, set the current live version as `v1.0.0`, and release this as `v1` and `v1.1.0`. That way, the previous version that uses the old form of the API is accessible.

**Review by bodhish — APPROVED** [review]
LGTM;

**Pull request merged**